tests/ssl-opt.sh

```diff
@@ -3875,6 +3875,8 @@ run_test    "Session resume using tickets: AES-256-CCM" \
             -s "a session has been resumed" \
             -c "a session has been resumed"
 
+requires_config_enabled MBEDTLS_CAMELLIA_C
+requires_config_enabled MBEDTLS_CCM_C
```

Comment:
Really? It seems to be this is supposed to work with drivers (with `USE_PSA` or course), so we should use a dependency declaration that reflects that. Or am I missing something?

Reply:
Yes, in theory you are right it should work, but the problem here is that `ssl_server2` uses `mbedtls_cipher_info_from_string()` to convert the `ticket_aead` to some cipher value. Internally, this goes to `cipher_wrap` which itself depends on legacy symbols.
In order to overcome this situation I will check if it's possible to mimic the `mbedtls_cipher_info_from_string` inside the `ssl_server/ssl_client` applications, so that the test is still possible without relying on `cipher/cipher_wrap` modules

Reply:
There's just one call to `mbedtls_cipher_info_from_string`. It's in a test program where we can change the syntax if we don't like the current one, and we don't care about the code size. It's only used for a limited subset (AEAD). So `ssl_server2.c` should just get its own string-to-id code, now that we want independence from `MBEDTLS_CIPHER_C`.

Reply:
Seconded. The primary purpose of `ssl_client2` is to support testing of SSL/TLS features, so we should do what it takes to make it possible to test everything that should work in driver-only builds.